src/optimizer/stats/table_stats.cpp

```diff
@@ -0,0 +1,161 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// table_stats.cpp
+//
+// Identification: src/optimizer/stats/table_stats.cpp
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "optimizer/stats/column_stats.h"
+#include "optimizer/stats/table_stats.h"
+#include "common/logger.h"
+
+namespace peloton {
+namespace optimizer {
+
+TableStats::TableStats(size_t num_rows,
+                       std::vector<std::shared_ptr<ColumnStats>> col_stats_ptrs)
+    : num_rows(num_rows),
+      col_stats_list_(col_stats_ptrs)
+{
+  for (size_t i = 0; i < col_stats_ptrs.size(); ++i) {
+    AddColumnStats(col_stats_ptrs[i]);
+  }
+}
+
+TableStats::TableStats(std::vector<std::shared_ptr<ColumnStats>> col_stats_ptrs)
+  : col_stats_list_(col_stats_ptrs)
+{
+  size_t col_count = col_stats_ptrs.size();
+  for (size_t i = 0; i < col_count; ++i) {
+    AddColumnStats(col_stats_ptrs[i]);
+  }
+  if (col_count == 0) {
+    num_rows = 0;
+  } else {
+    num_rows = col_stats_ptrs[0]->num_rows;
+  }
+}
+
+void TableStats::UpdateNumRows(size_t new_num_rows) { num_rows = new_num_rows; }
+
+bool TableStats::HasIndex(const std::string column_name) {
+  auto column_stats = GetColumnStats(column_name);
+  if (column_stats == nullptr) {

```

Comment:
shouldn't this be a PL_ASSERT instead of returning false. There is no way of distinguishing whether the column_stats was null or it did not have an index, since in both cases you return false.